lib/metrics/context.js

```diff
@@ -0,0 +1,44 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+var isA = require('joi')
+var HEX = require('../routes/validators').HEX_STRING
+
+module.exports = {
+  schema: isA.object({
+    flowId: isA.string().length(32).regex(HEX).required(),
+    flowBeginTime: isA.number().integer().positive().required(),
```

Comment:
I wonder about required-vs-optional for these flow properties, in cases where it's e.g. a native client rather than content-server making the calls.  ISTM we could just as easily create default values for them on the server as we could on the client, and the less work the client has to do to send us metrics the better.  But I don't feel terribly strongly either way.